lib/spack/spack/schema/config.py

```diff
@@ -19,6 +19,7 @@
     "config": {
         "type": "object",
         "default": {},
+        "additionalProperties": False,
```

Comment:
I think we shouldn't add this here, as we do remove config options that we don't want to fail on.